forge-game/src/main/java/forge/game/card/CardFactory.java

```diff
@@ -443,7 +443,37 @@ private static void readCardFace(Card c, ICardFace face) {
                 SpellAbility sa = new LandAbility(c);
                 sa.setCardState(c.getCurrentState());
                 c.addSpellAbility(sa);
-            } else if (c.isPermanent() && !c.isAura()) {
+            } else if (c.isAura()) {
+                String desc = "";
+                String extra = "";
+                for (KeywordInterface ki : c.getKeywords(Keyword.ENCHANT)) {
+                    String o = ki.getOriginal();
+                    String m[] = o.split(":");
+                    if (m.length > 2) {
+                        desc = m[2];
+                    } else {
+                        desc = m[1];
+                        if (CardType.isACardType(desc) || "Permanent".equals(desc) || "Player".equals(desc) || "Opponent".equals(desc)) {
+                            desc = desc.toLowerCase();
+                        }
+                    }
+                    break;
+                }
+                if (c.hasSVar("AttachAITgts")) {
+                    extra += " | AITgts$ " + c.getSVar("AttachAITgts");
+                }
+                if (c.hasSVar("AttachAILogic")) {
+                    extra += " | AILogic$ " + c.getSVar("AttachAILogic");
+                }
+                if (c.hasSVar("AttachAIValid")) { // TODO combine with AttachAITgts
+                    extra += " | AIValid$ " + c.getSVar("AttachAIValid");
+                }
+                String st = "SP$ Attach | ValidTgts$ Card,Player | TgtZone$ Battlefield,Graveyard | TgtPrompt$ Select target " + desc + extra;
```

Comment:
I don't understand how this is supposed to work
where do you restrict the targeting?

Reply:
Via `GameEntity.canBeEnchantedBy` where i check all Enchant Keywords (for corner cases)